core/carp_pattern.h

```diff
@@ -584,11 +584,11 @@ String Pattern_internal_add_value(PatternMatchState *ms, String res, String src,
         }
         res = Pattern_internal_add_char(res, tr[i]);
       }
-      else if (tr[i] == '0') res = String_append(res, src);
+      else if (tr[i] == '0') res = StringCopy_append(res, src);
```

Comment:
@hellerve Should probably start using `String_append` here and free `res` and `src` explicitly?

Reply:
Probably a good idea.